nf_core/components/nfcore_component.py

```diff
@@ -90,3 +94,47 @@ def _get_included_components(self, main_nf: str):
                 if line.strip().startswith("include"):
                     included_components.append(line.strip().split()[-1].split(self.org)[-1].split("main")[0].strip("/"))
         return included_components
+
+    def get_inputs_from_main_nf(self):
+        """Collect all inputs from the main.nf file."""
+        inputs = []
+        with open(self.main_nf, "r") as f:
+            data = f.read()
+        # get input values from main.nf after "input:", which can be formatted as tuple val(foo) path(bar) or val foo or val bar or path bar or path foo
+        # regex matches:
+        # val(foo)
+        # path(bar)
+        # val foo
+        # val bar
+        # path bar
+        # path foo
+        # don't match anything inside comments or after "output:"
+        if "input:" not in data:
+            log.debug(f"Could not find any inputs in {self.main_nf}")
+            return inputs
+        input_data = data.split("input:")[1].split("output:")[0]
+        regex = r"(val|path)\s*(\(([^)]+)\)|\s*([^)\s]+))"
```

Comment:
🤯 😍 
I tried this regex and I think group4 is including a comma if it is written right after the input name, see https://regex101.com/r/mtSt8q/1

Adding a comma in group 4 should fix it
```suggestion
        regex = r"(val|path)\s*(\(([^)]+)\)|\s*([^)\s,]+))"
```